lib/go/common/constants.go

```diff
@@ -0,0 +1,41 @@
+// **** DO NOT EDIT
+// **** This code is generated by the build process.
+
+package common
+
+// Constants for node attribtues
+const (
+	ARCH                = "si.io/arch"
+	HostName            = "si.io/hostname"
+	RackName            = "si.io/rackname"
+	OS                  = "si.io/os"
+	InstanceType        = "si.io/instance-type"
+	FailureDomainZone   = "si.io/zone"
+	FailureDomainRegion = "si.io/region"
+	LocalImages         = "si.io/local-images"
+	NodePartition       = "si.io/node-partition"
+)
+
+// Constants for allocation attribtues
+const (
+	ApplicationID  = "si.io/application-id"
+	ContainerImage = "si.io/container-image"
+	ContainerPorts = "si.io/container-ports"
+)
+// Cluster
+const DefaultNodeAttributeHostNameKey = "si.io/hostname"
+const DefaultNodeAttributeRackNameKey = "si.io/rackname"
+
+// Application
+const LabelApp = "app"
+const LabelApplicationID = "applicationId"
+const LabelQueueName = "queue"
+
+// Resource
+const Memory = "memory"
+const CPU = "vcore"
```

Comment:
Resources are untyped in the core. I don't think we should limit types or set description in the SI for that reason.

Reply:
I don't really get what you mean here. The string constants "memory" and "vcore" are used frequently in the core as well. IMO it makes sense to move it to the SI and use them in both the shim and the core.

Reply:
I think this is OK. @wilfred-s if you have more concerns, we can create another JIRA as a follow-up. At present, I think this is good to merge.